roxie/udplib/udptrr.cpp

```diff
@@ -339,6 +344,8 @@ class CReceiveManager : implements IReceiveManager, public CInterface
                 // Fall through
             case flowType::send_completed:
                 pendingRequests.append(requester);
+                returnPermits(requester->permitsIssued);
```

Comment:
I think this only needs to be in the previous case (ok_to_send) - might make the code clearer.